pkg/volume/util/operationexecutor/node_expander.go

```diff
@@ -100,9 +101,15 @@ func (ne *NodeExpander) runPreCheck() bool {
 		ne.pvcAlreadyUpdated = true
 	}
 
-	// PVC is already expanded but we are still trying to expand the volume because
-	// last recorded size in ASOW is older. This can happen for RWX volume types.
-	if ne.pvcAlreadyUpdated && storage.ContainsAccessMode(ne.pvc.Spec.AccessModes, v1.ReadWriteMany) {
+	// if the volume is already expanded, but volume is of type RWX and
+	// pvc doesn't have annotation indicating that node expansion is not required
+	// then we should allow node expansion to proceed, even if the volume is already expanded.
```

Comment:
Will there be any race between multiple node expansion?

Reply:
I do not see an easy way this could race with multiple node expansions, even if one node expansion was required and another one wasn't, since we don't start processing 2nd request until previous one finishes. 

Did you had a scenario in mind?